modules/drivers/mongo/src/metabase/driver/mongo/parameters.clj

```diff
@@ -139,6 +143,13 @@
           :else                        [(conj acc (substitute-field-filter v))
                                         missing]))
 
+      (params/ReferencedQuerySnippet? v)
+      (substitute-native-query-snippet [acc missing] v)
+
+      (params/ReferencedCardQuery? v)
```

Comment:
Since we haven't supported reference to saved questions like SQL, let's have a detailed error message here.

Reply:
- Is this the only unsupported parameter type?
- How does this compare with the way missing/unsupported parameters have been reported? On a superficial look, it seems we want to report all parameters we have problems with, but this change will eagerly report a problem about first question reference.

Reply:
> Is this the only unsupported parameter type? 
yes

> reporting error
Currently, we only have the report for missing required parameters. 
I also think this error is different in kind because if we have a referenced card parameter, we're not allowed to run it at all. 

Whereas with missing params, users could just provide values and it runs just fine.
So I think it's fine to leave it here.